crates/bevy_ecs/src/query/filter.rs

```diff
@@ -12,14 +12,13 @@ use std::{marker::PhantomData, ptr};
 
 // TODO: uncomment this and use as shorthand (remove where F::Fetch: FilterFetch everywhere) when
```

Comment:
You also need to remove the `F::Fetch: FilterFetch` everywhere.

Reply:
here for example: https://github.com/bevyengine/bevy/blob/b07db8462f20fff9f5f6c1af915ad73700b7a264/crates/bevy_ecs/src/query/iter.rs#L14